django/core/handlers/base.py

```diff
@@ -344,16 +344,13 @@ def check_response(self, response, callback, name=None):
 
     def make_view_atomic(self, view):
         non_atomic_requests = getattr(view, "_non_atomic_requests", set())
-        for db in connections.all():
-            if (
-                db.settings_dict["ATOMIC_REQUESTS"]
-                and db.alias not in non_atomic_requests
-            ):
+        for alias, settings_dict in connections.settings.items():
+            if settings_dict["ATOMIC_REQUESTS"] and alias not in non_atomic_requests:
```

Comment:
This is slightly different than before. Before the database wrapper could have changed the settings dict. I am not sure that would have been a good idea and is certainly not something we support. In hindsight for database pools where we will need a static settings dict anyways, this might be okay?

Reply:
IMO it's fine.